presto-benchmark-runner/src/main/java/com/facebook/presto/benchmark/framework/BenchmarkQueryExecutor.java

```diff
@@ -0,0 +1,140 @@
+import static com.facebook.presto.spi.StandardErrorCode.EXCEEDED_TIME_LIMIT;
+import static com.facebook.presto.sql.parser.ParsingOptions.DecimalLiteralTreatment.AS_DOUBLE;
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class BenchmarkQueryExecutor
+{
+    private static final Logger log = Logger.get(BenchmarkQueryExecutor.class);
+    public static final ParsingOptions PARSING_OPTIONS = ParsingOptions.builder().setDecimalLiteralTreatment(AS_DOUBLE).build();
+
+    private final PrestoActionFactory prestoActionFactory;
+    private final String testId;
+    private final SqlParser sqlParser;
+
+    @Inject
+    public BenchmarkQueryExecutor(
+            PrestoActionFactory prestoActionFactory,
+            BenchmarkRunnerConfig benchmarkRunnerConfig,
+            SqlParser sqlParser)
+    {
+        this.prestoActionFactory = requireNonNull(prestoActionFactory, "prestoAction is null");
+        this.testId = requireNonNull(benchmarkRunnerConfig, "testId is null").getTestId();
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+    }
+
+    public BenchmarkQueryEvent run(BenchmarkQuery benchmarkQuery, Map<String, String> sessionProperties)
+    {
+        QueryStats queryStats = null;
+        Statement statement = sqlParser.createStatement(benchmarkQuery.getQuery(), PARSING_OPTIONS);
+        PrestoAction prestoAction = prestoActionFactory.get(benchmarkQuery, sessionProperties);
+
+        try {
+            queryStats = prestoAction.execute(statement);
+            return buildEvent(benchmarkQuery, Optional.ofNullable(queryStats), Optional.empty());
+        }
+        catch (QueryException e) {
+            return buildEvent(benchmarkQuery, Optional.ofNullable(queryStats), Optional.of(e));
+        }
+        catch (Throwable t) {
+            log.error(t);
+            return buildEvent(benchmarkQuery, Optional.ofNullable(queryStats), Optional.empty());
+        }
+    }
+
+    private BenchmarkQueryEvent buildEvent(
+            BenchmarkQuery benchmarkQuery,
+            Optional<QueryStats> queryStats,
+            Optional<QueryException> queryException)
+    {
+        boolean succeeded = queryStats.isPresent();
+
+        QueryState state = getQueryState(queryStats, queryException);
+        String errorMessage = null;
+        if (!succeeded) {
+            errorMessage = format("Query:%s %s.\n\n", benchmarkQuery.getName(), state.name());
+
+            if (queryException.isPresent()) {
+                queryStats = queryException.get().getQueryStats();
+                errorMessage += format(
+                        "%s query with Queryid:%s failed on cluster:\n%s",
+                        benchmarkQuery.getName(),
+                        queryStats.map(QueryStats::getQueryId),
+                        getStackTraceAsString(queryException.get().getCause()));
+            }
+        }
```

Comment:
Unlike verification where the error message may also contains information whether run query runs are matched or mismatched, we don't need to construct an artificial error message here.
errorMessage should just be `queryException.getMessage()`, and we also need a field to store `getStackTraceAsString(queryException.getCause())`